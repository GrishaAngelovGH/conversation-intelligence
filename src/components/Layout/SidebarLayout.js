import PropTypes from 'prop-types'
import { Sidebar, Menu, MenuItem } from 'react-pro-sidebar'

const SidebarLayout = ({ children }) => (
  <div className="row g-0">
    <div className="col-md-2">
      <Sidebar width='210px'>
        <Menu>
          <MenuItem href='/' prefix={<span>Home</span>} icon={<i className="bi bi-house-door"></i>} />
          <MenuItem href='/transcript-summary' prefix={<span>Transcript Summary</span>} icon={<i className="bi bi-card-list"></i>} />
        </Menu>
      </Sidebar>
    </div>
    <div className="col-md-10">
      {
        children
      }
    </div>
  </div>
)

SidebarLayout.propTypes = {
  children: PropTypes.element.isRequired
}

export default SidebarLayout