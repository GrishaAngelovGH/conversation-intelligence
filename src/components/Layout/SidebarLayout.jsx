import { Link } from 'react-router-dom'
import { Sidebar, Menu } from 'react-pro-sidebar'

const SidebarLayout = ({ children }) => (
  <div className='row g-0'>
    <div className='col-md-2'>
      <Sidebar width='210px'>
        <Menu>
          <Link to='/' className='d-block border mt-1 ms-1 p-2 text-decoration-none bg-transparent text-primary rounded'>
            <i className='bi bi-house-door'></i> <span className='fw-bold'>Home</span>
          </Link>
          <Link to='/transcript-summary' className='d-block border ms-1 p-2 text-decoration-none bg-transparent text-primary rounded'>
            <i className='bi bi-card-list'></i> <span className='fw-bold'>Transcript Summary</span>
          </Link>
          <Link to='/seller-details' className='d-block border ms-1 p-2 text-decoration-none bg-transparent text-primary rounded'>
            <i className='bi bi-person'></i> <span className='fw-bold'>Seller Details</span>
          </Link>
        </Menu>
      </Sidebar>
    </div>
    <div className='col-md-10'>
      {
        children
      }
    </div>
  </div>
)

export default SidebarLayout