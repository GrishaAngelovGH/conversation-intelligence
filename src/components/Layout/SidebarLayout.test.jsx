import { render } from '@testing-library/react'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'

import SidebarLayout from './SidebarLayout'

test('should render SidebarLayout component', () => {
  const view = render(
    <Router>
      <Routes>
        <Route
          path='/'
          element={(
            <SidebarLayout>
              <div>content</div>
            </SidebarLayout>
          )} />
      </Routes>
    </Router>
  )

  expect(view).toMatchSnapshot()
})