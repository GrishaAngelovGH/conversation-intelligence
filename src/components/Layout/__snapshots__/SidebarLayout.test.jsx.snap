// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render SidebarLayout component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0"
      >
        <div
          class="col-md-2"
        >
          <aside
            class="ps-sidebar-root css-1p7obn9"
            data-testid="ps-sidebar-root-test-id"
            width="210px"
          >
            <div
              class="ps-sidebar-container css-dip3t8"
              data-testid="ps-sidebar-container-test-id"
            >
              <nav
                class="ps-menu-root css-vj11vy"
              >
                <ul
                  class="css-ewdv3l"
                >
                  <a
                    class="d-block border mt-1 ms-1 p-2 text-decoration-none bg-transparent text-primary rounded"
                    data-discover="true"
                    href="/"
                  >
                    <i
                      class="bi bi-house-door"
                    />
                     
                    <span
                      class="fw-bold"
                    >
                      Home
                    </span>
                  </a>
                  <a
                    class="d-block border ms-1 p-2 text-decoration-none bg-transparent text-primary rounded"
                    data-discover="true"
                    href="/transcript-summary"
                  >
                    <i
                      class="bi bi-card-list"
                    />
                     
                    <span
                      class="fw-bold"
                    >
                      Transcript Summary
                    </span>
                  </a>
                  <a
                    class="d-block border ms-1 p-2 text-decoration-none bg-transparent text-primary rounded"
                    data-discover="true"
                    href="/seller-details"
                  >
                    <i
                      class="bi bi-person"
                    />
                     
                    <span
                      class="fw-bold"
                    >
                      Seller Details
                    </span>
                  </a>
                </ul>
              </nav>
            </div>
          </aside>
        </div>
        <div
          class="col-md-10"
        >
          <div>
            content
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0"
    >
      <div
        class="col-md-2"
      >
        <aside
          class="ps-sidebar-root css-1p7obn9"
          data-testid="ps-sidebar-root-test-id"
          width="210px"
        >
          <div
            class="ps-sidebar-container css-dip3t8"
            data-testid="ps-sidebar-container-test-id"
          >
            <nav
              class="ps-menu-root css-vj11vy"
            >
              <ul
                class="css-ewdv3l"
              >
                <a
                  class="d-block border mt-1 ms-1 p-2 text-decoration-none bg-transparent text-primary rounded"
                  data-discover="true"
                  href="/"
                >
                  <i
                    class="bi bi-house-door"
                  />
                   
                  <span
                    class="fw-bold"
                  >
                    Home
                  </span>
                </a>
                <a
                  class="d-block border ms-1 p-2 text-decoration-none bg-transparent text-primary rounded"
                  data-discover="true"
                  href="/transcript-summary"
                >
                  <i
                    class="bi bi-card-list"
                  />
                   
                  <span
                    class="fw-bold"
                  >
                    Transcript Summary
                  </span>
                </a>
                <a
                  class="d-block border ms-1 p-2 text-decoration-none bg-transparent text-primary rounded"
                  data-discover="true"
                  href="/seller-details"
                >
                  <i
                    class="bi bi-person"
                  />
                   
                  <span
                    class="fw-bold"
                  >
                    Seller Details
                  </span>
                </a>
              </ul>
            </nav>
          </div>
        </aside>
      </div>
      <div
        class="col-md-10"
      >
        <div>
          content
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
